File: yt/yt/flow/tests/companion/types/python/pipeline/type_mapper.py
"""RowFunction that performs a straightforward identity mapping of every typed
field. It reads each column of the incoming message and writes the same value
to the output stream, exercising the companion wire-protocol round-trip for
each distinct value type (string, int64, uint64, double, boolean)."""

from yt.yt.flow.library.python.companion.computation import RowFunction

# Columns mirrored verbatim, one per distinct wire value type.
PAYLOAD_FIELDS = ("f_string", "f_int64", "f_uint64", "f_double", "f_bool")


class TypeMapper(RowFunction):
    def on_message(self, message, output, ctx):
        out = ctx.message_builder("mapped")
        out.set("key", message.payload["key"])
        for column in PAYLOAD_FIELDS:
            out.set(column, message.payload[column])
        output.add_message(out.finish())

File: yt/yt/flow/tests/companion/types/java/companion/src/main/java/tech/ytsaurus/flow/tests/types/TypeMapper.java
package tech.ytsaurus.flow.tests.types;

import tech.ytsaurus.flow.computation.OutputCollector;
import tech.ytsaurus.flow.context.RuntimeContext;
import tech.ytsaurus.flow.function.RowFunction;
import tech.ytsaurus.flow.row.ExtendedMessage;

/**
 * Performs a straightforward identity mapping of every typed field: reads each
 * column of the incoming message and writes the same value to the output
 * stream, exercising the companion wire-protocol round-trip for every distinct
 * value type (string, int64, uint64, double, boolean).
 */
public class TypeMapper implements RowFunction {

    @Override
    public void onMessage(ExtendedMessage message, OutputCollector output, RuntimeContext ctx) {
        output.addMessage(ctx.createMessageBuilder("mapped")
                .set("key", message.get("key", String.class))
                .set("f_string", message.get("f_string", String.class))
                .set("f_int64", message.get("f_int64", Long.class))
                .set("f_uint64", message.get("f_uint64", Long.class))
                .set("f_double", message.get("f_double", Double.class))
                .set("f_bool", message.get("f_bool", Boolean.class))
                .finish());
    }
}
